server/controllers/finance/reports/reportAccountsMultiple/index.js

```diff
@@ -0,0 +1,111 @@
+const _ = require('lodash');
+const q = require('q');
+const ReportManager = require('../../../../lib/ReportManager');
+
+const AccountsExtra = require('../../accounts/extra');
+const Exchange = require('../../exchange');
+const Currency = require('../../currencies');
+const accountExtra = require('../../accounts/extra');
+const db = require('../../../../lib/db');
+
+const TEMPLATE = './server/controllers/finance/reports/reportAccountsMultiple/report.handlebars';
+
+const { getAccountTransactions } = require('../reportAccounts/index');
+/**
+ * @method document
+ *
+ * @description
+ * Renders the PDF template for the Accounts Statement Report.
+ *
+ * The report contains the following information:
+ *  1. headers with the opening balance line and All general ledger transactions
+ *  for each account.  This opening balance line is
+ *  converted on the date of the `dateFrom` range.
+ *
+ */
+async function document(req, res, next) {
+  let report;
+  const bundle = {};
+
+  const params = req.query;
+  params.user = req.session.user;
+  params.enterprise_id = req.session.enterprise.id;
+  params.isEnterpriseCurrency = (req.session.enterprise.currency_id === Number(params.currency_id));
+
+  const accountSql = `SELECT id, type_id FROM account WHERE id=?`;
+
+  try {
+    report = new ReportManager(TEMPLATE, req.session, params);
+  } catch (e) {
+    return next(e);
+  }
+
+
+  params.dateFrom = (params.dateFrom) ? new Date(params.dateFrom) : new Date();
+  let accountTransactions = []; // list of all account containing transactions
+  const accountIDs = [].concat(params.accountIds);
+
+  const accounts = await q.all(accountIDs.map(id => {
```

Comment:
Nice use of `await`.  We could apply it to the entire function, but it isn't necessary here.